src/cpu.c

```diff
@@ -324,168 +349,228 @@ static int init(void) {
   return 0;
 } /* int init */
 
-/* Takes the zero-index number of a CPU and makes sure that the module-global
- * cpu_states buffer is large enough. Returne ENOMEM on erorr. */
-static int cpu_states_alloc(size_t cpu_num) /* {{{ */
-{
-  cpu_state_t *tmp;
-  size_t sz;
+static int usage_init(usage_t *u, cdtime_t now) {
+  if (u == NULL || now == 0) {
+    return EINVAL;
+  }
+
+  if (u->time != 0 && u->time < now) {
+    u->interval = now - u->time;
+  }
+  u->time = now;
+  u->cpu_num = 0;
+  u->finalized = false;
```

Comment:
For consistency with `usage_finalize()` move this to end of function (easier to notice).

Reply:
Done.